manifests/fact.pp

```diff
@@ -34,18 +34,38 @@
   $match = "^${name}=\\S.*$"
 
   if $facts_file != $facter::facts_file {
-    file { "facts_file_${name}":
-      ensure => file,
-      path   => $facts_file_path,
-      owner  => $facter::facts_file_owner,
-      group  => $facter::facts_file_group,
-      mode   => $facter::facts_file_mode,
+    if $facter::facts_file_purge {
+      $concat_target = "facts_file_${name}"
```

Comment:
I thought target is a path to a file, not an arbitrary string.

Reply:
It's the name of the `concat` resource to add the fragment, which is either `facts_file` from main class or `facts_file_${name}` from this location